ui/types/vault/app-types.ts

```diff
@@ -97,6 +97,35 @@ export type WithFormFieldsAndValidationsModel = WithFormFieldsModel & {
   validate(): ModelValidations;
 };
 
+// capabilities
+export interface Capabilities {
+  canCreate: boolean;
+  canDelete: boolean;
+  canList: boolean;
+  canPatch: boolean;
+  canRead: boolean;
+  canSudo: boolean;
+  canUpdate: boolean;
+}
+
+export interface CapabilitiesMap {
+  [key: string]: Capabilities;
```

Comment:
the key here would be the path tied to the capability, right? I know I built parts of this service, just trying to remember 😂

Reply:
Yes that's right. Since we don't know what the value might be we need to index it as a string.